src/test/test-strv.c

```diff
@@ -726,6 +726,16 @@ static void test_strv_fnmatch(void) {
         assert_se(strv_fnmatch(v, "\\", FNM_NOESCAPE));
 }
 
+static void test_strv_join_quoted_utf8(void) {
+        _cleanup_strv_free_ char **v = NULL;
+        _cleanup_free_ char *s = NULL;
+
+        /* test for issue #8056 */
+
+        assert_se(v = strv_new("/usr/bin/echo ", "привет", NULL));
+        assert_se(s = strv_join_quoted(v));
+}
```

Comment:
Initially, my solution looked like this as well. But I figured there should also be an assert that uses `streq` to check if you really got what you wanted from `strv_join_quoted`.

Reply:
Yeah. But, now `strv_join_quoted()` is just a function for tests in test-strv.c. Of course, more tests is better than fewer. But it may not be necessary here.